src/elf.rs

```diff
@@ -360,21 +365,28 @@ impl<'a, E: UserDefinedError, I: InstructionMeter> EBpfElf<E, I> {
 
         // calculate the text section info
         let text_section_info = SectionInfo {
+            name: elf
+                .shdr_strtab
+                .get(text_section.sh_name)
+                .unwrap()
+                .unwrap()
+                .to_string(),
             vaddr: text_section.sh_addr.saturating_add(ebpf::MM_PROGRAM_START),
             offset_range: text_section.file_range(),
         };
+        if text_section_info.vaddr > ebpf::MM_STACK_START {
```

Comment:
the overlap checks elsewhere won't catch this case?

Reply:
This checks the `vaddr`, other places check the file offset. Or is there a specific one that does that too?